mesonbuild/mesonlib.py

```diff
@@ -109,22 +118,31 @@ def check_direntry_issues(direntry_array):
         for de in direntry_array:
             if is_ascii_string(de):
                 continue
-            mlog.warning('''You are using {!r} which is not a Unicode-compatible '
-locale but you are trying to access a file system entry called {!r} which is
-not pure ASCII. This may cause problems.
-'''.format(e, de), file=sys.stderr)
+            mlog.warning(textwrap.dedent('''
+                You are using {!r} which is not a Unicode-compatible
+                locale but you are trying to access a file system entry called {!r} which is
+                not pure ASCII. This may cause problems.
+                '''.format(e, de)), file=sys.stderr)
+
 
 # Put this in objects that should not get dumped to pickle files
 # by accident.
 import threading
 an_unpicklable_object = threading.Lock()
 
+
 class MesonException(Exception):
     '''Exceptions thrown by Meson'''
 
+    file = None    # type: T.Optional[str]
+    lineno = None  # type: T.Optional[str]
```

Comment:
Just in case this is a mistake, and just from the names, I would expect `linono` and `colno` to be `int`s.

Reply:
You are correct, those should be ints.